### ISSUE TITLE
Cannot login with SSO -- 'continue' button is not working

### ISSUE THREAD

**Issue by banderlog** [ogra1/zoom-snap#58]
I enter  domain name when zoom asks for it, but the 'continue' button is not working

**Comment by banderlog** [comment]
I also tried:

```
EnableEmbedBrowserForSSO=true
forceSSOURL=%my_url%
sso_domain=%mu_domain%
```

The `/snap/zoom-client/113/.zoom/logs/zoom_stdout_stderr.log` is uninformative, it just loads and then exits gracefully. All button clicking process is out of its scope.

**Comment by banderlog** [comment]
[zoom-terminal.log](https://github.com/ogra1/zoom-snap/files/5549503/zoom-terminal.log)
[zoom_stdout_stderr.log](https://github.com/ogra1/zoom-snap/files/5549504/zoom_stdout_stderr.log)

**Comment by ogra1** [comment]
can you check again ? according to issue #39 it seems to work now

**Comment by banderlog** [comment]
nope

Tried both `sudo snap install zoom-client` and build it from current master with consequent installation.

**Comment by ogra1** [comment]
hmm, that''s sad, did you also try the workaround described in #39 ?

FYI, you don't need to build master yourself, it gets auto-built in the edge channel as soon as a change occurs, you can just:

    sudo snap install --edge zoom-client

to get the binary for the current master branch ...

**Comment by Fuseteam** [comment]
@banderlog is that with sso or with google login? it seemed work with google login for me, it opens the browser from where i can open it back in zoom

**Comment by banderlog** [comment]
@ogra1 You mean this workaround?
>Doing SSO login for work, I copy the "Launch Zoom" url link and run it on cli with xdg-open .

Where I should get a mentioned URL?

@Fuseteam both. It just does not open any browser. I have few and they are not snap-installed. Every other links works just fine.

**Comment by ogra1** [comment]
it definitely works fine with the google/facebook buttons for me, for the google account (i dont have a facebook account, but the correct page seems to open) it offers me to confirm and to take me back to zoom.

![Bildschirmfoto von 2020-11-28 20-51-21](https://user-images.githubusercontent.com/15386365/100524813-9433cc00-31bb-11eb-9b41-e521851b8b78.png)

what desktop environment are you running and what distro ?
(there seems to be a bug that it opens an additional empty tab, but it definitely opens the right SSO page alongside on ubuntu 16.04 and 20.04)

**Comment by Fuseteam** [comment]
i also open my browser for me on ubuntu 18.04, i think did set my default browser using xdg-mime iirc

**Comment by banderlog** [comment]
+ Ubuntu 20.04.1 LTS
+ i3-wm

Official zoom installation via deb package works as expected. As any other application with a need of URL opening in external browser.

But I didn't try to generate link with some other snap applications.

**Comment by ogra1** [comment]
do you have xdg-open installed on the host ?
snapd internally hands URLs (after mediating it) off to a daemon running in the user session, that in turn hands the URL to xdg-open in your desktop to run it in the preferred application found in the mime database ...

**Comment by banderlog** [comment]
>do you have xdg-open installed on the host ?

Yes, and it works, e.g.: `xdg-open https://www.google.com` will open link in firefox.

I even have installed `snapd-xdg-open` as a separate package and then tried again (with reinstallation) -- nothing changed

**Comment by MarkusLohmayer** [comment]
Please have a look at my recent answer/question. Maybe it solves your problem: https://askubuntu.com/questions/1309660/snap-xdg-open-does-nohing/1310176#1310176

**Comment by oscarkey** [comment]
I'm hitting the same problem, when using the Sway desktop environment:
Linux 5.11.14-arch1-1, sway 1.5.1-2, snapd 2.49.1-1, zoom-client 5.6.1-13632.0328

To summarise what happens for me (which appears to be the same as above):
- Nothing happens when clicking "continue" after entering the domain when signing in with SSO in the Zoom client
- Even if I set `embeddedBrowserForSSOLogin=true` in `~/snap/zoom-client/current/.config/zoomus.conf`
- `xdg-open https://google.com` does work from the shell launched using `snap run --shell zoom-client`

If it's helpful, I was able to work around the problem by:
- installing the non-snap Zoom client alongside the snap version
- starting SSO sign in from inside the non-snap client, up to the point where it opens the login page in Firefox
- uninstalling the non-snap Zoom client
- finishing signing in in Firefox, which then opens the snap Zoom client
